gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/DagProcUtils.java

```diff
@@ -174,12 +171,6 @@ public static void cancelDagNode(Dag.DagNode<JobExecutionPlan> dagNodeToCancel,
         log.warn("No Job future when canceling DAG node - {}", dagNodeToCancel.getValue().getId());
       }
       DagManagerUtils.getSpecProducer(dagNodeToCancel).cancelJob(dagNodeToCancel.getValue().getJobSpec().getUri(), cancelJobArgs).get();
-      // add back the dag node with updated states in the store
```

Comment:
addDag will happen only by LaunchDagProc once in the life cycle of Dag
and
updateDagNode will happen only after submitting the job to SpecProducer (to store the future) or in Reevaluate to update the status

Reply:
updating the status too soon (like here), may cause a dag to delete before ReevaluateDagProc is processed.